lib/util.rb

```diff
@@ -18,4 +19,37 @@ def self.rootish_ssh(host, user, keys, cmd)
       ret
     end
   end
+
+  def self.create_certificate(subject:, duration:, extensions: [], issuer_cert: nil, issuer_key: nil)
+    cert = OpenSSL::X509::Certificate.new
+    key = OpenSSL::PKey::EC.generate("prime256v1")
+
+    # If the issuer is nil, we will create a self-signed certificate.
+    if issuer_cert.nil?
+      issuer_cert = cert
+      issuer_key = key
+    end
+
+    # Set certificate details
+    cert.version = 2 # X.509v3
+    cert.serial = OpenSSL::Random.random_bytes(16).unpack1("Q>*")
```

Comment:
Huh, I somehow don't remember having to do this, I think OpenSSL may have a way to get an openssl "BN" (bignum) that can be interpreted by the `serial` field directly. But I could be misremembering.

Reply:
Changed it to BN.rand.